screens/Profile/UpdateMyProfile/UpdateMyProfileScreen.tsx

```diff
@@ -1,10 +1,107 @@
-import { View, Text } from "react-native"
+import React, { useEffect, useState } from "react"
+import { View, Text, Image, Pressable, ActivityIndicator } from "react-native"
 import { styles } from "./styles"
+import { RouteProp, useRoute, useNavigation } from "@react-navigation/native"
+import { User } from "../../../types/User"
+import { launchImageLibraryAsync, MediaTypeOptions } from 'expo-image-picker'
+import { updateUserData, updateUserImage, uploadUserImageToStorage } from "../../../firebase/User"
+import { peach } from "../../../assets/colors/colors"
+import InputField from "../../../components/InputField/InputField"
+import { showSuccessToast } from "../../../components/ToastMessage/toast"
+
+type RootStackParamList = {
+  UpdateProfile: {
+      user: User
+  }
+}
+
+type MapScreenRouteProp = RouteProp<RootStackParamList, 'UpdateProfile'>;
 
 export const UpdateMyProfileScreen = () => {
+  const route = useRoute<MapScreenRouteProp>()
+  const navigation = useNavigation()
+  const user = route.params.user
+  const [image, setImage] = useState(user.profilePicture)
+  const [imageLoading, setImageLoading] = useState(false)
+  const [loading, setLoading] = useState(false)
+  const [name, setName] = useState(user.firstName)
+  const [lastName, setLastName] = useState(user.lastName)
+  const [location, setLocation] = useState(user.location)
+
+  const pickImage = async () => {
+    const result = await launchImageLibraryAsync({
+      mediaTypes: MediaTypeOptions.All,
+      allowsEditing: true,
+      aspect: [3, 3],
+      quality: 1,
+    })
+
+    if (!result.canceled) {
+      setImageLoading(true)
+      const url = await uploadUserImageToStorage(user.uid, result.assets[0].uri)
+      if (url) {
+        await updateUserImage(user.uid, url)
+        setImage(url)
+      }
+      setTimeout(() => setImageLoading(false), 500)
+    }
+  }
+
+  useEffect(() => {
+    setLoading(true)
+    // TODO: Later
+    setLoading(false)
+  }, [])
+
+  const submitChanges = async () => {
+    setLoading(true)
+    const newData = {
+      firstName: name,
+      lastName: lastName,
+      location: location
+    }
+    const success = await updateUserData(user.uid, newData)
+    if(success) {
+      showSuccessToast("Profile updated successfully!")
+      navigation.goBack()
+    }
+    setLoading(false)
+  }
+
+
   return (
     <View style={styles.container}>
-      <Text>This is the update my profile screen</Text>
+      <Image style={styles.image} source={{uri: image}}/>
```

Comment:
When not having a profile picture, I get this warning: 

<img src="https://github.com/uniconnect-epfl/uniconnect/assets/93329823/a30627bb-4ee0-4cb7-b991-540027ec999f"  height="600">

I believe you could use the ProfilePicture component (you may need to pull main) to fix this warning. (There may be a mistake in the component because pictureUrl ? does not account for empty strings but you can modify that)

Reply:
fixed!